tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/TestScenarioRunner.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.hawkular.apm.tests.dockerized.environment.DockerComposeExecutor;
+import org.hawkular.apm.tests.dockerized.environment.DockerImageExecutor;
+import org.hawkular.apm.tests.dockerized.environment.TestEnvironmentExecutor;
+import org.hawkular.apm.tests.dockerized.exception.TestFailException;
+import org.hawkular.apm.tests.dockerized.model.JsonPathVerify;
+import org.hawkular.apm.tests.dockerized.model.TestCase;
+import org.hawkular.apm.tests.dockerized.model.TestScenario;
+import org.hawkular.apm.tests.dockerized.model.Type;
+import org.hawkular.apm.tests.server.ApmMockServer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.docker.client.exceptions.DockerCertificateException;
+
+/**
+ * Class for running test scenario.
+ *
+ * @author Pavol Loffay
+ */
+public class TestScenarioRunner {
+
+    /**
+     * Version of APM, needed for ENV variable inside docker image
+     */
+    private final String apmVersion;
+    /**
+     * Port at which will be listening APM server
+     */
+    private final int apmServerPort;
+
+    /**
+     * Mocked Hawkular APM server to collect information captured by agents.
+     */
+//    private final ApmMockServer apmServer;
+    private final ObjectMapper objectMapper;
+
+    private final ApmMockServer apmServer;
+
+    /**
+     * @param apmVersion - version of the Hawkular APM (usually current version)
+     * @param apmServerPort - port at which Hawkular APM server will be started
+     * @throws DockerCertificateException
+     */
+    public TestScenarioRunner(String apmVersion, int apmServerPort) throws DockerCertificateException {
+        this.apmVersion = apmVersion;
+        this.apmServerPort = apmServerPort;
+
+        this.objectMapper = new ObjectMapper();
+        apmServer = new ApmMockServer();
+        apmServer.setPort(apmServerPort);
+        apmServer.setShutdownTimer(60*60*1000);
```

Comment:
1 hour? What does this effectively mean? That the server will force a shutdown after 1 hour?